crypto/pkcs11/pkcs11helpers.go

```diff
@@ -163,10 +169,8 @@ func rsaPublicEncryptOAEP(pubKey *rsa.PublicKey, plaintext []byte) ([]byte, stri
 // pkcs11UriGetLoginParameters gets the parameters necessary for login from the Pkcs11URI
 // PIN and module are mandatory; slot-id is optional and if not found -1 will be returned
 func pkcs11UriGetLoginParameters(p11uri *pkcs11uri.Pkcs11URI) (string, string, int64, error) {
-	pin, err := p11uri.GetPIN()
-	if err != nil {
-		return "", "", 0, errors.Wrap(err, "No PIN available in pkcs11 URI")
-	}
+	// public key access may not require a PIN
+	pin, _ := p11uri.GetPIN()
```

Comment:
Should there be a way to check if the error is related to this exception? go 1.13 has a ` errors.Is` / `errors.As` function, not sure if that would be useful here.

Reply:
You mean an error later on that occurs when the decryption failed due to no private key being accessible, which in turn may be due to a missing PIN? Maybe in this case we could show in the error message that 'no PIN was used'. Or we could require a PIN in this function here if a private key operation is to be done and return a hard error if we could not get a PIN.

Reply:
The latter may be better assuming all private key ops on all pkcs11 devices require a PIN.

Reply:
yea either way, i was actually more thinking about the `GetPin` function itself. I feel like there are scenarios where "a PIN cannot be obtained because there is no PIN" and "the PIN can't be obtained because of an unexpected failure", in the later case we would perhaps want to stop and propagate the error? Is there a way to distinguish between these?

Reply:
Do think a function like `HasPin` would help resolve this overload-issue? I don't think comparing error codes is good and having an additional boolean return code could be confusing as well.

Reply:
I implemented `HasPIN` and I am using it here now. I added a patch to update the go-pkcs11uri dependency.